### PR TITLE
add BulkPutWithStreamStrategyExample

### PR DESCRIPTION
**Pull request by shabtaisharon**

### CODE REVIEW

File: Ds3Examples/BulkPutWithStreamStrategyExample.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                new Credentials(
+                    ConfigurationManager.AppSettings["Ds3AccessKey"],
+                    ConfigurationManager.AppSettings["Ds3SecretKey"]
+                )
+            ).Build();
+
+            // Set up the high-level abstractions.
+            IDs3ClientHelpers helpers = new Ds3ClientHelpers(client);
+
+            const string bucket = "bucket-name";
```
Comment (DenverM80) [inline Ds3Examples/BulkPutWithStreamStrategyExample.cs]: i prefer to use a bucket name like "put-bulk-with-stream-test" so its easier to follow in logs and look at if it ever regresses

File: Ds3Examples/BulkPutWithStreamStrategyExample.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            // Creates a bulk job with the server based on the files in a directory (recursively).
+            var directoryObjects = FileHelpers.ListObjectsForDirectory(directory).ToList();
+            var job = helpers.StartWriteJob(bucket, directoryObjects, helperStrategy: new WriteStreamHelperStrategy());
+
+            // Tracing example
+            if (ClientSwitch.TraceInfo) { Trace.WriteLine(string.Format("StartWriteJob({0})", bucket)); }
+            if (ClientSwitch.TraceVerbose) { Trace.WriteLine(string.Format("Add files from: {0}", directory)); }
+
+            // Keep the job id around. This is useful for job recovery in the case of a failure.
+            Console.WriteLine("Job id {0} started.", job.JobId);
```
Comment (DenverM80) [inline Ds3Examples/BulkPutWithStreamStrategyExample.cs]: is there no other log mechanism for users to override?

File: Ds3Examples/BulkPutWithStreamStrategyExample.cs (hunk 1 above)
Comment (shabtaisharon) [inline Ds3Examples/BulkPutWithStreamStrategyExample.cs]: This is not a test, it is an example.
but i'll change the name...

File: Ds3Examples/BulkPutWithStreamStrategyExample.cs (hunk 2 above)
Comment (shabtaisharon) [inline Ds3Examples/BulkPutWithStreamStrategyExample.cs]: nop

File: Ds3Examples/BulkPutWithStreamStrategyExample.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            }
+
+            foreach (var md5 in md5s)
+            {
+                Console.WriteLine("Done transferring file {0} with MD5 value {1}", md5.Key, Convert.ToBase64String(md5.Value.Hash));
+            }
+
+            // Wait for user input.
+            Console.WriteLine("Press enter to continue.");
+            Console.ReadLine();
```
Comment (DenverM80) [inline Ds3Examples/BulkPutWithStreamStrategyExample.cs]: I see, this is an example not a test - safe to ignore my previous question

**Pull request merged**